app/models/search_query.rb

```diff
@@ -64,13 +64,13 @@ class SearchQuery < ApplicationRecord
   belongs_to :parent, class_name: 'SearchQuery'
   has_many   :children, class_name: 'SearchQuery'
 
-  validates_presence_of :search_text, if: :search?
+  validates_presence_of :search_text, if: :search_query_type?
 
   enum query_type: {
       search: 'search',
       filter: 'filter',
       list: 'list'
-  }
+  },  _suffix: true
```

Comment:
This change is due to the following reason 
`from /Users/yikangmao/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/activerecord-5.2.4.5/lib/active_record/enum.rb:226:in `detect_enum_conflict!'
/Users/yikangmao/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/activerecord-5.2.4.5/lib/active_record/enum.rb:235:in `raise_conflict_error': You tried to define an enum named "search_query_type" on the model "SearchQuery", but this will generate a class method "filter", which is already defined by ActiveRecord::Relation. (ArgumentError)
`